ISL.Providers.PDS.FakeFHIR/Services/Foundations/PdsService.cs

```diff
@@ -4,29 +4,156 @@
 
 using Hl7.Fhir.Model;
 using ISL.Providers.PDS.Abstractions.Models;
+using ISL.Providers.PDS.FakeFHIR.Mappers;
+using ISL.Providers.PDS.FakeFHIR.Models;
 using System;
+using System.Collections.Generic;
+using System.Linq;
 using System.Threading.Tasks;
 
 namespace ISL.Providers.PDS.FakeFHIR.Services.Foundations
 {
-    internal class PdsService : IPdsService
+    internal partial class PdsService : IPdsService
     {
+        private readonly FakeFHIRProviderConfigurations fakeFHIRProviderConfiguration;
+        private List<PdsPatientDetails> FakePatientDetails = new List<PdsPatientDetails>();
+
+        public PdsService(
+            FakeFHIRProviderConfigurations fakeFHIRProviderConfiguration)
+        { 
+            this.fakeFHIRProviderConfiguration = fakeFHIRProviderConfiguration;
+            this.FakePatientDetails = GetFakePatientDetails();
+        }
+
         public ValueTask<PatientBundle> PatientLookupByDetailsAsync(string givenName = null,
             string familyName = null,
             string gender = null,
-            string postCode = null,
+            string address = null,
             string dateOfBirth = null,
             string dateOfDeath = null,
             string registeredGpPractice = null,
             string email = null,
-            string phoneNumber = null)
+            string phoneNumber = null) =>
+            TryCatch(async () =>
+            {
+                List<PdsPatientDetails> filteredPatients = FilterPatientByDetails(givenName,
+                    familyName,
+                    gender,
+                    address,
+                    dateOfBirth,
+                    dateOfDeath, registeredGpPractice,
+                    email,
+                    phoneNumber);
```

Comment:
If you pass in the `this.FakePatientDetails` then your method will be easier to test with random data.

```suggestion
                List<PdsPatientDetails> filteredPatients = FilterPatientByDetails(givenName,
                    familyName,
                    gender,
                    address,
                    dateOfBirth,
                    dateOfDeath, registeredGpPractice,
                    email,
                    phoneNumber,
                    this.FakePatientDetails);
```